pkg/cmd/status/status.go

```diff
@@ -0,0 +1,616 @@
+	Number     int
+	Repository struct {
+		NameWithOwner string
+	}
+}
+
+type Results []SearchResult
+
+func (rs Results) Len() int {
+	return len(rs)
+}
+
+func (rs Results) Less(i, j int) bool {
+	return rs[i].UpdatedAt.After(rs[j].UpdatedAt)
+}
+
+func (rs Results) Swap(i, j int) {
+	rs[i], rs[j] = rs[j], rs[i]
+}
+
+type StatusGetter struct {
+	Client         *http.Client
+	cachedClient   func(*http.Client, time.Duration) *http.Client
+	Org            string
+	Exclude        string
+	AssignedPRs    []StatusItem
+	AssignedIssues []StatusItem
+	Mentions       []StatusItem
+	ReviewRequests []StatusItem
+	RepoActivity   []StatusItem
+}
+
+func NewStatusGetter(client *http.Client, opts *StatusOptions) *StatusGetter {
+	return &StatusGetter{
+		Client:       client,
+		Org:          opts.Org,
+		Exclude:      opts.Exclude,
+		cachedClient: opts.CachedClient,
+	}
+}
+
+func (s *StatusGetter) CachedClient(ttl time.Duration) *http.Client {
+	return s.cachedClient(s.Client, ttl)
+}
+
+func (s *StatusGetter) ShouldExclude(repo string) bool {
+	return strings.Contains(s.Exclude, repo)
+}
+
+func (s *StatusGetter) CurrentUsername() (string, error) {
+	cachedClient := s.CachedClient(time.Hour * 48)
+	cachingAPIClient := api.NewClientFromHTTP(cachedClient)
+	currentUsername, err := api.CurrentLoginName(cachingAPIClient, ghinstance.Default())
+	if err != nil {
+		return "", fmt.Errorf("failed to get current username: %w", err)
+	}
+
+	return currentUsername, nil
+}
+
+func (s *StatusGetter) ActualMention(n Notification) (string, error) {
+	currentUsername, err := s.CurrentUsername()
+	if err != nil {
+		return "", err
+	}
+
+	// long cache period since once a comment is looked up, it never needs to be
+	// consulted again.
+	cachedClient := s.CachedClient(time.Hour * 24 * 30)
+	c := api.NewClientFromHTTP(cachedClient)
+	resp := struct {
+		Body string
+	}{}
+	if err := c.REST(ghinstance.Default(), "GET", n.Subject.LatestCommentURL, nil, &resp); err != nil {
+		return "", err
+	}
+
+	var ret string
+
+	if strings.Contains(resp.Body, "@"+currentUsername) {
+		ret = resp.Body
+	}
+
+	return ret, nil
+}
+
+// These are split up by endpoint since it is along that boundary we parallelize
+// work
+
+// Populate .Mentions
+func (s *StatusGetter) LoadNotifications() error {
+	perPage := 100
+	c := api.NewClientFromHTTP(s.Client)
+	query := url.Values{}
+	query.Add("per_page", fmt.Sprintf("%d", perPage))
+	query.Add("participating", "true")
+	query.Add("all", "true")
+
+	// this sucks, having to fetch so much :/ but it was the only way in my
+	// testing to really get enough mentions. I would love to be able to just
+	// filter for mentions but it does not seem like the notifications API can
+	// do that. I'd switch to the GraphQL version, but to my knowledge that does
+	// not work with PATs right now.
+	var ns []Notification
+	var resp []Notification
+	pages := 3
+	for page := 1; page <= pages; page++ {
+		query.Add("page", fmt.Sprintf("%d", page))
```

Comment:
This results in the `page` param being repeated: `?page=1&page=2&per_page=100`

Tip: there is no need to manually construct pagination URLs; the API response after getting the first page will include the `link` response header that will contain the full URL to the next page of results https://github.com/cli/cli/blob/058c2b4fe3938465662ecb5d8fdcb1060087cc51/pkg/cmd/api/pagination.go#L13-L22